### PR TITLE
Update tooltip-popper to handle smaller screensizes.

### PR DESCRIPTION
**Pull request by catandthemachines**
## Summary:
Adjusting implementation of Popper in tooltip-popper to handle smaller screen sizes by:
- Adjusting location of popper based on document size, not viewport size. This will ensure the popper doesn't default to inaccessible locations when the popper doesn't fit in the viewport size.
- Disabled isReferenceHidden flag to ensure the popper stays present even when the reference anchor is out of view. This will allow users in higher zoom levels or smaller screen sizes to scroll to interact with the popper if necessary instead of it disappearing when the reference isn't in view.

Issue: LEMS-2085

## Test plan:
- Go to url: https://khan.github.io/wonder-blocks/iframe.html?args=&id=packages-popover-popover--with-actions&viewMode=story
- Set the page body to have padding: "0 0 500px 0" to ensure the page body is larger than the height of the button example.
- Set the viewport size to 320 x 256
- You should see the popover attach at the bottom of the trigger element since there is more room within the page document.

### CODE REVIEW

**Comment by jandrade** [comment]
@catandthemachines This approach seems to have a counter effect. When the Tooltip/Popover is overflowing the viewport, the tooltip is not repositioning itself and instead it appears cut off.

You can see an example of that here:

https://5e1bf4b385e3fb0020b7073c-ocepdnhrkc.chromatic.com/?path=/docs/packages-popover-popover--docs#with%20actions

Screenshot 2024-07-12 at 5 11 38 PM

While this fixes this behavior for screen sizes, it will introduce some unexpected behaviors like the one in the screenshot.

**Comment by catandthemachines** [comment]
> @catandthemachines This approach seems to have a counter effect. When the Tooltip/Popover is overflowing the viewport, the tooltip is not repositioning itself and instead it appears cut off.
> 
> You can see an example of that here:
> 
> https://5e1bf4b385e3fb0020b7073c-ocepdnhrkc.chromatic.com/?path=/docs/packages-popover-popover--docs#with%20actions
> 
> Screenshot 2024-07-12 at 5 11 38 PM
> While this fixes this behavior for screen sizes, it will introduce some unexpected behaviors like the one in the screenshot.

Yes this was intentional as it appears, just my own opinion, to be better as the user can scroll to see the full content. Is this a requirement for Popover? I will be meeting with our accessibility designer to ask their opinion and any concerns she might have.

**Comment by jandrade** [comment]
@catandthemachines Updating here to set expectations. As per our last conversation, this change mostly makes sense, but we still want to keep the `flip` functionality around so the Popover dialog is still visible in some situations.

File: packages/wonder-blocks-tooltip/src/components/tooltip-popper.tsx
Comment (handeyeco) [inline packages/wonder-blocks-tooltip/src/components/tooltip-popper.tsx:230]: I'm confused by the use of `Partial`. Are we saying that only part of the `PreventOverflowModifier` will be used or are we trying to say that these modifiers are optional?

Like should it be:

``` TS
type Modifier = PreventOverflowModifier | FlipModifier | SmallViewportModifier;
const modifiers: Modifier[] = [smallViewportModifier];
```

Likely I'm just confused.

File: packages/wonder-blocks-tooltip/src/components/tooltip-popper.tsx
Comment (handeyeco) [inline packages/wonder-blocks-tooltip/src/components/tooltip-popper.tsx:101]: This doesn't need to use `setState` or anything? I'm guessing not, I'm guessing it's Popper state and not React state. Just thought I'd double check.

**Review by handeyeco — COMMENTED** [review]
Looking good. I'll defer to FEI/WB folks to approve though since I know there's been some back and forth.

File: packages/wonder-blocks-tooltip/src/components/tooltip-popper.tsx
Comment (handeyeco) [inline packages/wonder-blocks-tooltip/src/components/tooltip-popper.tsx:81]: I've never seen `object` used like this before. Is this basically the `any` of objects?

**Comment by jandrade** [comment]
@catandthemachines would you mind adding a Storybook story explaining the use case for this new prop + including the padding necessary for recreating this experience? This one doesn't need to be included in the Chromatic snapshots.

**Comment by catandthemachines** [comment]
> @catandthemachines would you mind adding a Storybook story explaining the use case for this new prop + including the padding necessary for recreating this experience? This one doesn't need to be included in the Chromatic snapshots.

Added a new storybook page, you can find it in Chromatic at `/?path=/story/packages-popover-popover--with-document-root-boundary`

File: __docs__/wonder-blocks-popover/popover.stories.tsx
Comment (jandrade) [inline __docs__/wonder-blocks-popover/popover.stories.tsx:768]: **suggestion:** Could you please update this description to match the story? Probably the popover content would be useful for this.

File: packages/wonder-blocks-tooltip/src/components/tooltip-popper.tsx
Comment (jandrade) [inline packages/wonder-blocks-tooltip/src/components/tooltip-popper.tsx:235]: **suggestion:** Could you please add a TODO comment here pointing to the floating-ui Jira ticket? this would be helpful for us to track this change in the future.

```suggestion
        // TODO(WB-1680): Use floating-ui's 
        const modifiers: Modifiers[] = [smallViewportModifier];
```

File: packages/wonder-blocks-tooltip/src/components/tooltip-popper.tsx
Comment (jandrade) [inline packages/wonder-blocks-tooltip/src/components/tooltip-popper.tsx:233]: **suggestion:** I'd recommend to move this modifier definition at the the file level, that way the `_modifyPosition` function won't be recreated on every render.

**Review by jandrade — APPROVED** [review]
Looks good! I've left some suggestions to tackle before landing. Thanks for iterating over this solution! 🚢

**Pull request merged**
